include/mega/sync.h

```diff
@@ -41,6 +44,38 @@ int computeReversePathMatchScore(string& accumulated, const string& path1, const
 bool assignFilesystemIds(Sync& sync, MegaApp& app, FileSystemAccess& fsaccess, handlelocalnode_map& fsidnodes,
                          const string& localdebris, const string& localseparator);
 
+// A collection of unsyncable remote nodes stored by node handle
+class MEGA_API UnsyncableNodeBag
+{
+public:
+    UnsyncableNodeBag() = default;
+
+    UnsyncableNodeBag(DbAccess& dbaccess, FileSystemAccess& fsaccess, PrnGen& rng);
+
+    MEGA_DISABLE_COPY_MOVE(UnsyncableNodeBag)
+
+    // Adds a new node by handle and sync path
+    bool addNode(handle nodeHandle, const std::string& syncPath);
+
+    // Removes a node by handle
+    bool removeNode(handle nodeHandle);
+
+    // Returns the sync path of the given node handle if the node is not syncable, null otherwise
+    const std::string* syncPath(handle nodeHandle) const;
+
+private:
+    uint32_t mNextTableId = 0; // the next table ID to use
+
+    struct NodeData
+    {
+        std::string mSyncPath; // the sync that the node belongs to
```

Comment:
Storing a path per unsyncable node could make the data structure quite large in RAM - eg million node account, user manually removes most of the destination nodes.   Could we get away with just testing the presence/absence of the nodehandle in the set, rather than by path?

Reply:
Removed syncpath.